src/machine_logrotate.py

```diff
@@ -0,0 +1,48 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Logrotate implementation for machine charm"""
+
+import logging
+
+import jinja2
+
+import logrotate
+
+logger = logging.getLogger(__name__)
+
+CHARMED_MYSQL_COMMON_DIRECTORY = "/var/snap/charmed-mysql/common"
+SYSTEM_USER = "snap_daemon"
+ROOT_USER = "root"
```

Comment:
no, because cron needs `sudo` access to be able to execute as user

Reply:
cron doesnt, but the `logrotate` needs to run as `root` -> i have tested the file permissions with `snap_daemon` or `ubuntu`, and logrotate doesnt fire unless these files are `root`

Reply:
the issue is that we have a logrotate config (from the spec) that specifies `su snap_daemon snap_daemon -> for logrotate to be able to change the user, it need to be `root` (or at least `sudoer`)

Reply:
well, the default user for the machine is `ubuntu` and not `snap_daemon`. since `su` requires sudo privileges to be able to change the user/group, the file needs to be run with `root`. if the file is owned by not `root`, then it will be unable to access the file

Reply:
`logrotate` is run by a cron (neither are run in the context of the snap). sorry, there is no role for the `ubuntu` user

1. cron jobs are run as the root user -> this is why the cron file needs to be owned by `root` ([reference](https://tecadmin.net/running-a-crontab-as-www-data-user-in-linux/))
2. logrotate needs to be run as root user (so it can run `su`) -> this is why the logrotate config file is owned by `root`
3. logrotate will de-escalate down to `snap_daemon` so the log rotate will be run as `snap_daemon`

Reply:
we will let cron run as root for now, but the logrotate process will be run as `snap_daemon`. we will also investigate replacing cron with systemd timers in the near future